clients/java/signalr/src/main/java/HubConnection.java

```diff
@@ -43,36 +47,41 @@ public HubConnection(String url, Transport transport) {
             HubMessage[] messages = protocol.parseMessages(payload);
 
             for (HubMessage message : messages) {
+                logger.log(LogLevel.Debug,"Received message of type %s%n", message.getMessageType());
```

Comment:
Ah, I see, you're putting `%n` in the log messages themselves. I don't think we should require that the caller provide the newlines. The newline isn't part of the "log message" it's part of what the Console Logger has to do in order to render it to the console. If we were logging to a database or some other storage, you probably wouldn't want a newline here